### PR TITLE
Update docs for GA

### PR DESCRIPTION
**Pull request by cdhoffmann**
Updates the docs with the new APIs and changes to existing APIs in the Usage Documentation. Removes the BETA acknowledgement section from the readme in prep for GA.

### CODE REVIEW

File: Documentation/Usage/MobileCore.md [hunk 1]
```diff
@@ -291,13 +305,29 @@ AEPEvent *event = [[AEPEvent alloc] initWithName:@"My Event" type:AEPEventType.c
...
+##### Dispatching an `Event` with a timeout:
+
+###### Swift
+
+```swift
+let event = Event(name: "My Event", type: EventType.custom, source: EventType.custom, data: ["exampleKey": "exampleVal"])
+MobileCore.dispatch(event: event, timeout: 2) { (responseEvent) in
+    // handle responseEvent
+}```
+
```
Comment (shalehaha) [inline Documentation/Usage/MobileCore.md:317]: `###### Objective-C`

File: Documentation/Usage/MobileCore.md [hunk 2]
```diff
@@ -378,7 +408,7 @@ MobileCore.getSdkIdentities { (ids, error) in
 ###### Objective-C
 
 ```objective-c
-[AEPMobileCore getSdkIdentities:^(NSString * _Nullable ids, enum AEPError error) {
+[AEPMobileCore getSdkIdentities:^(NSString * _Nullable ids, NSError* _Nullable error) {
```
Comment (shalehaha) [inline Documentation/Usage/MobileCore.md:411]: There are several places in the Identity's docs stilling use `AEPError`, can you update them as well?

**Comment by yangyansong-adbe** [comment]
@cdhoffmann  
https://github.com/adobe/aepsdk-core-ios#cocoapods
can we also update `Installation - cocoapods` section to use version `3.0.0` not the git url and branch name?

https://github.com/adobe/aepsdk-core-ios#requirements
and also update `requirements` section like what we did in AEPRulesEngine repo https://github.com/adobe/aepsdk-rulesengine-ios#requirements

**Review by yangyansong-adbe — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -31,18 +25,14 @@ use_frameworks!
...
 # for app development, include all the following pods
 target 'YOUR_TARGET_NAME' do
-    pod 'AEPServices', :git => 'https://github.com/adobe/aepsdk-core-ios.git', :branch => 'main'
-    pod 'AEPCore', :git => 'https://github.com/adobe/aepsdk-core-ios.git', :branch => 'main'
-    pod 'AEPLifecycle', :git => 'https://github.com/adobe/aepsdk-core-ios.git', :branch => 'main'
-    pod 'AEPIdentity', :git => 'https://github.com/adobe/aepsdk-core-ios.git', :branch => 'main'
-    pod 'AEPRulesEngine', :git => 'https://github.com/adobe/aepsdk-rulesengine-ios.git', :branch => 'main'
+    pod 'AEPCore'
+    pod 'AEPLifecycle'
+    pod 'AEPIdentity'
```
Comment (yangyansong-adbe) [inline README.md:30]: Hi @cdhoffmann, @swarna04  just pointed out that we need to add the `AEPSignal` here.

**Pull request merged**